db/config.cc

```diff
@@ -574,8 +574,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
         "Threshold for commitlog disk usage. When used disk space goes above this value, Scylla initiates flushes of memtables to disk for the oldest commitlog segments, removing those log segments. Adjusting this affects disk usage vs. write latency. Default is (approximately) commitlog_total_space_in_mb - <num shards>*commitlog_segment_size_in_mb.")
     , commitlog_use_o_dsync(this, "commitlog_use_o_dsync", value_status::Used, true,
         "Whether or not to use O_DSYNC mode for commitlog segments IO. Can improve commitlog latency on some file systems.\n")
-    , commitlog_use_hard_size_limit(this, "commitlog_use_hard_size_limit", value_status::Used, true,
-        "Whether or not to use a hard size limit for commitlog disk usage. Default is true. Enabling this can cause latency spikes, whereas the default can lead to occasional disk usage peaks.\n")
+    , commitlog_use_hard_size_limit(this, "commitlog_use_hard_size_limit", value_status::Deprecated, true,
```

Comment:
btw it seems that `value_status::Deprecated` makes the parameter be ignored when read from yaml file, but it can still be set by the command line parameter, although the help prompt says it's ignored.

Reply:
In that case we should make sure that the value of `commitlog_use_hard_size_limit` is not used anywhere in the code, substituting the current uses with just `true`. That's e.g. what was done in https://github.com/scylladb/scylladb/pull/16334.

Reply:
done